### ISSUE TITLE
MktmpdirNode error

### ISSUE THREAD

**Issue by antonio-fiol** [DeanCording/node-red-contrib-fs-ops#7]
Using the node to create a temporary directory, I get
```"TypeError: "options" must be a string or an object"```

Investigating, I found that:
```
> fs.mkdtempSync("/tmp/test1")
'/tmp/test11Bv1sh'
> fs.mkdtempSync("/tmp/test2",0o777)
TypeError: "options" must be a string or an object, got number instead.
    at getOptions (fs.js:75:11)
    at Object.fs.mkdtempSync (fs.js:1861:13)
> fs.mkdtempSync("/tmp/test2",{"mode":0o777})
'/tmp/test2ooeuEi'
> fiol@antonio-T440:/tmp$ ls -ld test*
drwx------ 2 fiol fiol   4096 nov 28 09:40 test11Bv1sh
drwx------ 2 fiol fiol   4096 nov 28 09:40 test2ooeuEi
```
So:
1. mkdtempSync does not like a number as second parameter, at all.
2. Passing an object with a `mode` property gets ignored.
This all agrees with the documentation in https://nodejs.org/api/fs.html#fs_fs_mkdtempsync_prefix_options

Tried with node v8.12.0 and v6.14.4.
node v10.13.0 gives a very similar error, with slightly different wording:
```
> fs.mkdtempSync("/tmp/test2",0o777)
TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be one of type string or Object. Received type number
    at getOptions (internal/fs/utils.js:178:11)
    at Object.mkdtempSync (fs.js:1673:13)
```

Possible solutions I see:
a. Remove the if(fs.mkdtempSync) and keep the implementation based on mkdir only.
b. Remove the mode parameter from this node, and trim the code accordingly.
c. Use fs.chmodSync(path, mode). But beware! When you do fs.mkdirSync(path, mode), your umask is taken into account. If you use fs.chmodSync(path, mode), it does NOT take umask into account, so you end up with the real permissions you set. So... the behavior would not be consistent. Maybe you want to:
```
pathname = fs.mkdtempSync(path);
fs.chmodSync(pathname, mode & ~process.umask() );
```

**Comment by DeanCording** [comment]
Thanks for the great report!

I'll look into it but it may take me a week or so before I have the time.

**Comment by DeanCording** [comment]
Fix applied in Version 1.5.0

I went with option b as it matched the functionality of mktmpdir in node.fs

**Comment by antonio-fiol** [comment]
Thanks!
